### PR TITLE
Fix: Use $0 env variable to retrieve current active shell

### PR DESCRIPTION
**Pull request by imransolanki**
## What type of PR is this?

- bug

## What this PR does / why we need it:
This is a fix to [bug- Shell completions are broken when login shell is different than currently active shell #1884
](https://github.com/urfave/cli/issues/1884)
- Using `$0` as an env variable instead of `SHELL` as `$0` gives the currently active shell

## Which issue(s) this PR fixes:
[Fixes the bug- Shell completions are broken when a login shell is different than currently active shell #1884
](https://github.com/urfave/cli/issues/1884)

## Testing
Updated the existing test case.

## Release Notes
```release-note
Fixed https://github.com/urfave/cli/issues/1884
```

### CODE REVIEW

**Comment by bartekpacia** [comment]
Hey, can you tell us more about how it works?

**Comment by imransolanki** [comment]
> Hey, can you tell us more about how it works?

The code retrieves env var `$0` which gives us the active shell. 
Earlier the code was retrieving the current shell using `$SHELL` which denotes the login shell.
In the best-case scenario $SHELL and $0 are equal

Example of `$0` in action
image

**Review by bartekpacia — APPROVED** [review]
I see. Thank you, this change LGTM. I'm holding off with merge for other maintainers to take a look.

**Review by asahasrabuddhe — APPROVED** [review]
Looks good to me!

**Comment by bartekpacia** [comment]
I think this is worth backporting, yeah

**Review by dearchap — APPROVED** [review]
@imransolanki Thanks !!!!

**Comment by bartekpacia** [comment]
This actually doesn't fix anything, but breaks printing descriptions in `zsh`. This commit must be reverted.

I think this is also a lesson for us, maintainers of this module, to not trust that contributors did actually test their changes.

### Before this PR

https://github.com/user-attachments/assets/35856f1e-22df-4d2d-b397-846e9259ac7b

### After this PR

https://github.com/user-attachments/assets/b7b332f1-0bef-44b9-b468-ca1ae1127913

**Pull request merged**
